provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -233,32 +233,66 @@ func (c *Client) VirtualMachines(ctx context.Context, path string) ([]*mo.Virtua
 	return vms, nil
 }
 
-// ComputeResources returns list of all root compute resources in the system.
-func (c *Client) ComputeResources(ctx context.Context) ([]*mo.ComputeResource, error) {
+// ComputeResources returns a slice of all compute resources in the datacenter,
+// along with a slice of each compute resource's full path.
+func (c *Client) ComputeResources(ctx context.Context) ([]*mo.ComputeResource, []string, error) {
```

Comment:
I think maybe better to define a small computeResource struct with the resource and path and return a single slice of those

Reply:
Much better idea -- thanks